src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -311,7 +313,7 @@ public Solution WithProjectOutputFilePath(ProjectId projectId, string outputFile
         /// <summary>
         /// Creates a new solution instance with the project specified updated to have the reference assembly output file path.
         /// </summary>
-        public Solution WithProjectOutputRefFilePath(ProjectId projectId, string outputRefFilePath)
+        public Solution WithProjectOutputRefFilePath(ProjectId projectId, string? outputRefFilePath)
```

Comment:
This makes me think it would be better to have an overload instead of allowing null for the ref path? Just looking at the API with the `?` now looks more odd...

Reply:
Projects may not have output paths; that's a normal thing. I'm not sure what the concern is here?

Reply:
It's more of a meta gripe than anything. Now that we can annotate reference types with nullability, having a null string here is more akin to an implementation detail of _not_ having an output path. 

```C#
public Solution WithProjectOutputRefFilePath(ProjectId projectId, string outputFilePath);

public Solution WithProjectwithNoOutputRefFilePath(ProjectId projectId)
```

The terrible naming aside, the above has a more meaningful call pattern. We either want a path, or we want to make it so the project doesn't have one. Null may be how we rationalize that, but it's just a well known magic value. It doesn't tell you, for example, if that ends up succeeding and having some default value when you pass in null. 

Not worth blocking, for sure, just more of a thought about the meaning of null to represent absence or clearing of a value from an API perspective.